src/server/services/wms/qgswmsrenderer.cpp

```diff
@@ -2881,6 +2883,24 @@ namespace QgsWms
               feature.setGeometry( QgsGeometry::fromWkt( wkt ) );
             }
           }
+
+          // Note: this is the feature expression display name, not the field alias
+          if ( withDisplayName )
+          {
+            QString displayName;
+            const QDomNodeList attrs = featureNode.elementsByTagName( "Attribute" );
+            for ( int k = 0; k < attrs.count(); k++ )
+            {
+              const QDomElement elm = attrs.at( k ).toElement();
+              if ( elm.attribute( QStringLiteral( "name" ) ).compare( "displayName" ) == 0 )
+              {
+                displayName = elm.attribute( "value" );
+                break;
+              }
+            }
+            fidDisplayNameMap.insert( feature.id(), displayName );
```

Comment:
Is it possible that for a given feature displayName wouldn't be set ? If so, would we report an empty string, or nothing ? Not sure what is the desired behavior, and if that matters...

Besides that, LGTM

Reply:
I think it's possible even if probably rare, the display name comes from the evaluation of an user-defined expression.